sources/vault.move

```diff
@@ -0,0 +1,122 @@
+/// User can deposit/withdraw multiple fungible assets to/from different markets
+module playground::vault {
+    use std::signer;
+    use std::string;
+
+    use aptos_std::table::{Self, Table};
+    use aptos_framework::fungible_asset::{Self, FungibleAsset, Metadata};
+    use aptos_framework::object::{Self, Object};
+
+    #[test_only]
+    use aptos_framework::primary_fungible_store;
+    #[test_only]
+    use std::option;
+    use std::vector;
+
+    const ERR_UNAUTHORIZED: u64 = 1;
+    const ERR_INSUFFICIENT_FUNDS: u64 = 2;
+
+    /// Market holds user deposited assets
+    struct Market has key {
+        assets: Object<FungibleAsset>
+    }
+
+    /// Balance keeps track of user assets in different markets
+    struct User has key {
+        markets: Table<Object<Market>, u64>
+    }
+
+    public fun init_market(admin: &signer, asset: Object<Metadata>): Object<Market> {
+        assert!(signer::address_of(admin) == @playground, ERR_UNAUTHORIZED);
+
+        let asset_name = fungible_asset::name(asset);
+
+        let fa_seed = vector::empty<u8>();
+        vector::append(&mut fa_seed, b"fa-");
+        vector::append(&mut fa_seed, *string::bytes(&asset_name));
+        let (fa_signer, fa_obj) = create_object<FungibleAsset>(admin, fa_seed);
```

Comment:
You need to call fungible_asset::create_store to create a store of the fungible asset instead